### PR TITLE
Functional tests - Filter and sort files

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Filter and sort files
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/03_catalog/06_files/03_filterAndSortFiles.js" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19088)

### CODE REVIEW

File: tests/puppeteer/campaigns/functional/BO/03_catalog/06_files/03_filterAndSortFiles.js [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+    it('should reset all filters', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'resetFilterFirst', baseContext);
+      numberOfFiles = await this.pageObjects.filesPage.resetAndGetNumberOfLines();
+      await expect(numberOfFiles).to.be.above(0);
+    });
+  });
+  // 2 : Filter files table
+  describe('Filter file in BO', async () => {
+    const tests = [
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/03_catalog/06_files/03_filterAndSortFiles.js]:
```suggestion
    const filterTests = [
```

File: tests/puppeteer/campaigns/functional/BO/03_catalog/06_files/03_filterAndSortFiles.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+          testIdentifier: 'filterSize', filterType: 'input', filterBy: 'file_size', filterValue: '64',
+        },
+      },
+      {
+        args: {
+          testIdentifier: 'filterProducts', filterType: 'input', filterBy: 'products', filterValue: '1',
+        },
+      },
+    ];
+    tests.forEach((test) => {
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/03_catalog/06_files/03_filterAndSortFiles.js]:
```suggestion
    filterTests.forEach((test) => {
```

File: tests/puppeteer/campaigns/functional/BO/03_catalog/06_files/03_filterAndSortFiles.js [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+      it('should reset all filters', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', 'resetAfterDelete', baseContext);
+        const numberOfFilesAfterReset = await this.pageObjects.filesPage.resetAndGetNumberOfLines();
+        await expect(numberOfFilesAfterReset).to.be.equal(numberOfFiles);
+      });
+    });
+  });
+  // 3 : Sort files table
+  describe('Sort files', async () => {
+    const brandsTests = [
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/03_catalog/06_files/03_filterAndSortFiles.js]:
```suggestion
    const sortTests = [
```

File: tests/puppeteer/campaigns/functional/BO/03_catalog/06_files/03_filterAndSortFiles.js [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+      },
+      {args: {testIdentifier: 'sortByProductsAsc', sortBy: 'products', sortDirection: 'asc'}},
+      {args: {testIdentifier: 'sortByProductsDesc', sortBy: 'products', sortDirection: 'desc'}},
+      {
+        args: {
+          testIdentifier: 'sortByIdAsc', sortBy: 'id_attachment', sortDirection: 'asc', isFloat: true,
+        },
+      },
+    ];
+    brandsTests.forEach((test) => {
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/03_catalog/06_files/03_filterAndSortFiles.js]:
```suggestion
    sortTests.forEach((test) => {
```

File: tests/puppeteer/campaigns/functional/BO/03_catalog/06_files/03_filterAndSortFiles.js (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/puppeteer/campaigns/functional/BO/03_catalog/06_files/03_filterAndSortFiles.js]: Done thx :)

**Review by SimonGrn — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**